### PR TITLE
Provide Corpus to OSS-Fuzz via fuzz.sh

### PR DESCRIPTION
**Pull request by zi0Black**
### Description
This PR adds support to download custom corpus for our fuzzer running on OSS-Fuzz infrastructure.
Proper documentation updated.

### Test Plan
I used a local OSS-Fuzz build and run to test instructions posted [here](https://google.github.io/oss-fuzz/getting-started/new-project-guide/#seed-corpus).

### CODE REVIEW

**Review by GotenJBZ — APPROVED** [review]

**Comment by zi0Black** [comment]
Landed some additional changes.
 - Debug command to start a fuzzer with rust-dgb and a testcase
 - Logic to apply custom patches via git
 - A patch to speed-up fuzzing in single core environments
 - Disabled replay featured in `aptosvm_publish_and_run.rs`
 - Small fixes to the script

File: testsuite/fuzzer/fuzz/fuzz_targets/move/aptosvm_publish_and_run.rs [hunk 1]
```diff
@@ -420,7 +420,8 @@ fn run_case(mut input: RunnableState) -> Result<(), Corpus> {
     // exec tx
     tdbg!("exec start");
     let mut old_res = None;
-    const N_EXTRA_RERUNS: usize = 3;
+    const N_EXTRA_RERUNS: usize = 0;
```
Comment (gedigi) [inline testsuite/fuzzer/fuzz/fuzz_targets/move/aptosvm_publish_and_run.rs:423]: why this change? isn't this used to catch determinism issues?

Comment (zi0Black) [reply testsuite/fuzzer/fuzz/fuzz_targets/move/aptosvm_publish_and_run.rs:423]: I still think it is not the right place to test for them, execution in the fuzzer is really important and re-executing the same transaction multiple times slows down the process. As I mentioned multiple times the replay tool should check for this or we  run the fuzzer with this feature enabled on another instance. My2cent

**Review by gedigi — APPROVED** [review]
